### ISSUE TITLE
Golink from Docker not accessible from tailscale

### ISSUE THREAD

**Issue by mullender** [tailscale/golink#11]
I am executing this command
```
docker run -it -v --rm -e TS_AUTHKEY='tskey-auth-kc<>Jn' -v /tmp/golink:/root ghcr.io/tailscale/golink:main  -sqlitedb /root/test.db -verbose
```
The output looks good: `Serving http://go/ ...`

I can resolve the hostname:
```
dscacheutil -q host -a name go
name: go.tig<>.ts.net
ip_address: 100.<>.217.100
```

But when I attempt to visit http://go the connection times out:
```
curl go
curl: (28) Failed to connect to go port 80 after 75007 ms: Operation timed out
```

I must be holding it wrong :)

**Comment by bradfitz** [comment]
Please share a `tailscale bugreport` from the machine you're running `curl` on just before and/or after the curl. 

What does `tailscale ping go` say?

**Comment by bradfitz** [comment]
(I suspect this is going to be more of a docker question than a golink or Tailscale question, though.)

**Comment by mullender** [comment]
It is not a big deal, I can work around it.
But because I was following the instructions at https://github.com/tailscale/golink#running-in-production I thought I would report the issue, in case others run into it.

```
➜  ~ tailscale ping go
Warning: client version "1.32.3-t9dd89b8c2" != tailscaled server version "1.32.3-ta07555f43-g093d1e978"
ping "100.76.217.100" timed out
ping "100.76.217.100" timed out
ping "100.76.217.100" timed out
ping "100.76.217.100" timed out
^[[Aping "100.76.217.100" timed out
^C
➜  ~ ping go
PING go.tigon-micro.ts.net (100.76.217.100): 56 data bytes
Request timeout for icmp_seq 0
```

```
➜  ~ tailscale bugreport
Warning: client version "1.32.3-t9dd89b8c2" != tailscaled server version "1.32.3-ta07555f43-g093d1e978"
BUG-d681174189372a1e4d516bb4adc2f66a422cf7f415bf8675ed2726ff782cbce5-20221201055039Z-0844eb1802c42ad8
➜  ~
➜  ~ tailscale ping go
Warning: client version "1.32.3-t9dd89b8c2" != tailscaled server version "1.32.3-ta07555f43-g093d1e978"
ping "100.76.217.100" timed out
ping "100.76.217.100" timed out
^C
➜  ~ tailscale bugreport
Warning: client version "1.32.3-t9dd89b8c2" != tailscaled server version "1.32.3-ta07555f43-g093d1e978"
BUG-d681174189372a1e4d516bb4adc2f66a422cf7f415bf8675ed2726ff782cbce5-20221201055108Z-d878379c02af2f94
```

**Comment by parkr** [comment]
I needed to add a Search Domain to my network settings so "go" would expand to "go.mytailnet.ts.net". After that, it resolved fine in any browser and in the terminal.

**Comment by bradfitz** [comment]
@parkr, if MagicDNS is on (the default) and you're not disabling `--accept-dns` it should do that automatically.

@mullender, I see your go node stopping & starting like 7 times. Was that intentional or was it crashing on something? Do docker logs have some crash?

**Comment by mullender** [comment]
When I created the bug report I was running two identical docker nodes in parallel, by mistake (it was late). I am going to retrace my steps later. 

The go hostname resolves to a Tailscale IP as expected. 
The golink server inside the container says it is listening on port 80. 

When I try to connect to http://go:80/ the request fails.

**Comment by mullender** [comment]
I think I figured out what is going on. There is another 'go' host on the internal network. When I am connected to tailscale and the internal network at the same time, the browser resolves 'go' to the different ip address, but ping will resolve to the tailscale one.

I will check out the golink code and configure a different/unique hostname, to completely confirm this.

Feel free to close this.

**Comment by mullender** [comment]
I tested with the freshly added `-hostname` flag and picked a unique hostname and that resolved the issue completely.

**Comment by mullender** [comment]
closed by pr #14
